### DISCUSSION TITLE
What happens when one of the multiple Hocuspocus nodes is permanently offline?

### DISCUSSION THREAD

**Discussion by akirarika** [ueberdosis/hocuspocus#246] [Q&A]
I am a student from China. I have some questions. Before that, please allow me to express my gratitude to you for creating these great open source software. 🤗 my dream is to be like you.

this is what I understand (may not be correct) 😣:

If I want user data not to be lost after reboot, I need to use RocksDB extension.

If I want to scale horizontally, I need to enable Redis extension and connect all Hotuspocus nodes to the same Redis center / cluster.

If I understand correctly 🤔:

If the hard disk of one of my nodes is damaged and permanently offline, will the user's data be lost?

If a Hotuspocus node is added later, and the RocksDB is empty, and other nodes have accumulated a very very lot of data before, will Redis be filled?

Does the RocksDB in each Hocuspocus node store complete user data?

I see that Hocuspocus will launch the Postgres extension in the future. Is it an alternative to RocksDB? Does it allow all nodes to connect to the same database and the same table?

[TiDB](https://en.pingcap.com/) is very popular recently. It is an open source database compatible with MySQL protocol, and it can scale horizontally. In China, almost all large Internet companies are using it.

The name of the extension is Postgres, which means that it only supports PostgreSQL and will not support MySQL and TiDB? Let me guess, maybe it's because you want to use `jsonb`?

I'm not very good at the server. Maybe, my question is stupid. I hope I'm not disturbing you 😥

**Comment by Alecyrus** [comment]
There is an extension called [extension-database](https://github.com/ueberdosis/hocuspocus/tree/main/packages/extension-database)...
